tobler/dasymetric/raster_tools.py

```diff
@@ -27,6 +31,45 @@ def _apply_parser(df):
     return df.apply(_parse_geom)
 
 
+def _fast_append_profile_in_gdf(geodataframe, raster_path, force_crs_match=True):
+    """Append categorical zonal statistics (counts by pixel type) as columns to an input geodataframe.
+
+    geodataframe : geopandas.GeoDataFrame
+        geodataframe that has overlay with the raster. If some polygon do not overlay the raster,
+        consider a preprocessing step using the function subset_gdf_polygons_from_raster.
+    raster_path : str
+        path to the raster image.
+    force_crs_match : bool, Default is True.
+        Whether the Coordinate Reference System (CRS) of the polygon will be reprojected to
+        the CRS of the raster file. It is recommended to let this argument as True.
+
+    Notes
+    -----
+    The generated geodataframe will input the value 0 for each Type that is not present in the raster
+    for each polygon.
+    """
+
+    _check_presence_of_crs(geodataframe)
+    if force_crs_match:
+        with rio.open(raster_path) as raster:
+            # raster =
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```

Comment:
Why do we catch warnings here? If there's a specific waring we want to suppress, we should catch that one. Not all.

Reply:
i'm not sure why that's there. 

Agree, removed